Ballot SMC017v2: Increase Minimum RSA CA Key Size (#304)
cabforum/smime: SBR.md

| 1.0.8 | SMC010 | SHALL implement MPIC | May 15, 2025 |
| 1.0.12 | SMC014 | SHALL implement DNSSEC for CAA | March 15, 2026 |
| 1.0.14 | SMC016 | SHALL sunset all remaining use of SHA-1 in Certificates and CRLs | September 15, 2026 |
| 1.0.15  | SMC017   |New Root or Subordinate CA RSA Key SHALL be at least 4096 bits | September 15, 2026 |

> | 1.0.15  | SMC017   |SHALL cease issuance from Subordinate CA with RSA Key less than 4096 bits | September 15, 2027 |

Author: ```suggestion
| 1.0.15  | SMC017   |SHALL cease issuance from Subordinate CA with RSA Key less than 3072 bits | September 15, 2027 |
```

> | 1.0.15  | SMC017   |SHALL cease issuance from Subordinate CA with RSA Key less than 3072 bits | September 15, 2027 |

Author: ```suggestion
| 1.0.15  | SMC017   |SHALL cease Subscriber Certificate issuance from Subordinate CA with RSA Key less than 3072 bits | September 15, 2027 |
```

For RSA key pairs the CA SHALL:

* For Keys corresponding to Root and Subordinate CA Certificates signed on or after September 15, 2026 ensure that the modulus size, when encoded, is at least 4096 bits; and
* For Keys corresponding to Subscriber Certificates, ensure that the modulus size, when encoded, is at least 2048 bits; and
* Ensure that the modulus size, in bits, is evenly divisible by 8.

Effective September 15, 2027 the CA SHALL NOT issue Certificates from any Subordinate CA when:

> - the `tbsCertificate` contains `id-kp-emailProtection` in the `extKeyUsage` extension; and

Reviewer: I know the common style is to 'and' the requirements, but in this case, 'or' should be used because failing to fulfill any one of the requirements should block issuance.

Author: @aww-aww  I have to disagree with that. Using "or", would prevent the issuance of SubCAs, including Cross-signed CAs with a key size of less than 4096 bits (as those would require the EKU to be present). 

Since the SBRs themselves are already scoped to leaf certificates with id-kp-emailProtection + an email SAN, the proposed language seems appropriate?

Reviewer: I don't think I'm fully following your intent. I get the cross-sign issue, but is the intent to have requirements be the union of all the requirements or for any one of them be exclusionary aside from the cross-signs with smaller key sizes?

Author: Closing this as we discussed it on the previous call

For RSA key pairs the CA SHALL:

> * For Keys corresponding to Root and Subordinate CA Certificates signed on or after September 15, 2026 ensure that the modulus size, when encoded, is at least 4096 bits; and

Author: ```suggestion
* For Keys corresponding to CA Certificates (including Root, Subordinate and Cross Certificates) signed on or after September 15, 2026 ensure that the modulus size, when encoded, is at least 4096 bits; and
```

* For Keys corresponding to Subscriber Certificates, ensure that the modulus size, when encoded, is at least 2048 bits; and
* Ensure that the modulus size, in bits, is evenly divisible by 8.

> Effective September 15, 2027 the CA SHALL NOT issue Certificates from any CA when:

Author: ```suggestion
Effective September 15, 2027 the CA SHALL NOT issue Certificates from any CA if all of the following conditions are true:
```
